scripts/spack/packages/caliper/package.py

```diff
@@ -1,193 +1,178 @@
-            description='Enable stack unwind support')
-    variant('libdw', default=is_linux,
-            description='Enable DWARF symbol lookup')
+    variant(
+        "libunwind", default=sys.platform != "darwin", description="Enable stack unwind support"
+    )
+    variant("libdw", default=is_linux, description="Enable DWARF symbol lookup")
     # pthread_self() signature is incompatible with PAPI_thread_init() on Mac
-    variant('papi', default=sys.platform != 'darwin',
-            description='Enable PAPI service')
-    variant('libpfm', default=False,
-            description='Enable libpfm (perf_events) service')
+    variant("papi", default=sys.platform != "darwin", description="Enable PAPI service")
+    variant("libpfm", default=False, description="Enable libpfm (perf_events) service")
     # Gotcha is Linux-only
-    variant('gotcha', default=is_linux,
-            description='Enable GOTCHA support')
-    variant('sampler', default=is_linux,
-            description='Enable sampling support on Linux')
-    variant('sosflow', default=False,
-            description='Enable SOSflow support')
-    variant('fortran', default=False,
-            description='Enable Fortran support')
+    variant("gotcha", default=is_linux, description="Enable GOTCHA support")
+    variant("sampler", default=is_linux, description="Enable sampling support on Linux")
+    variant("sosflow", default=False, description="Enable SOSflow support")
+    variant("fortran", default=False, description="Enable Fortran support")
 
-    depends_on('adiak@0.1:0', when='@2.2: +adiak')
+    depends_on("adiak@0.1:0", when="@2.2: +adiak")
 
-    depends_on('papi@5.3:5', when='@:2.2 +papi')
-    depends_on('papi@5.3:6', when='@2.3: +papi')
+    depends_on("papi@5.3:5", when="@:2.2 +papi")
+    depends_on("papi@5.3:6", when="@2.3: +papi")
 
-    depends_on('libpfm4@4.8:4', when='+libpfm')
+    depends_on("libpfm4@4.8:4", when="+libpfm")
 
-    depends_on('mpi', when='+mpi')
-    depends_on('unwind@1.2:1', when='+libunwind')
-    depends_on('elfutils', when='+libdw')
+    depends_on("mpi", when="+mpi")
+    depends_on("unwind@1.2:1", when="+libunwind")
+    depends_on("elfutils", when="+libdw")
 
-    depends_on('sosflow@spack', when='@1.0:1+sosflow')
+    depends_on("sosflow@spack", when="@1.0:1+sosflow")
 
-    depends_on('cmake',  type='build')
-    depends_on('python', type='build')
+    depends_on("cmake", type="build")
+    depends_on("python", type="build")
 
     # sosflow support not yet in 2.0
-    conflicts('+sosflow', '@2.0.0:2.5')
-    conflicts('+adiak', '@:2.1')
-    conflicts('+libdw', '@:2.4')
+    conflicts("+sosflow", "@2.0.0:2.9")
+    conflicts("+adiak", "@:2.1")
+    conflicts("+libdw", "@:2.4")
+    conflicts("+rocm", "@:2.7")
+    conflicts("+rocm+cuda")
 
-    patch('for_aarch64.patch', when='target=aarch64:')
-    # SERAC EDIT START
-    patch('no_static_cupti.patch', when='+cuda~shared')
```

Comment:
@white238 do we still need this patch?

Reply:
It should be in the newer versions of caliper.

Reply:
It looks like our spack.yml says that libunwind is unbuildable. It may be that we need a more recent version.

Reply:
https://github.com/libunwind/libunwind/blob/master/src/x86_64/getcontext.S#L38

Reply:
Yea its there:
```
[quartz:~]$ nm -D /usr/lib64/libunwind.so | grep -i getcontext
0000000000005534 T _Ux86_64_getcontext
```